packages/automation-kit/README.md

```diff
@@ -2,10 +2,130 @@
+
+---
+
+
+## Automation-kit Architecture
+
+The `automation-kit` provides a set of high-level functions to interact with smart contracts, designed for seamless creation and management of Ditto automation workflows.
+
+
+### `automation-kit` API
+
+#### SmartWalletFactory (`VaultFactoryABI.json`)
+
+- **`predictVaultAddress`**  
+- **`deploy`**
+- **`getVersionAndIdByAddress`**
+
+
+#### SmartWallet (`VaultABI.json`)
+
+- **`deposit`**
+- **`withdraw`**
+- **`upgrade`**
+- **`executeOperation`**
+
+
+#### WorkflowFactory (`VaultABI.json`)
+
+- **`createWorkflow (buildAndDeploy)`** -> addWorkflowAndGelatoTask
+
+---
+
+
+### `web3-adapter-core` API
+
+The `web3-adapter-core` serves as the abstraction layer for interacting with smart contracts. It will handle tasks such as:
+1. **Building transactions** (`contractCall`), interacting with contracts, signing, and sending transactions.
+2. Abstracts the complexities of interacting with the blockchain, whether it's via **ethers.js**, **web3.js**, or **viem**.
+
+- **`use`** — Initializes the Web3Adapter with a specific library.
+- **`contractCall`**
+- **`sendTransaction`**
+- **`signTransaction`**
+- **`estimateGas`**
+
+---
+
+
+### How **automation-kit** and **web3-adapter-core** Work Together:
+
+1. **automation-kit** contains the logic for interacting with smart contracts, including ABI and methods such as `predictVaultAddress`, `deploy`, `deposit`, etc.
+2. **web3-adapter-core** abstracts the blockchain interaction. All calls that require network communication, such as transaction building and sending, will be done through **web3-adapter-core** methods (`contractCall`, `sendTransaction`, etc.).
+3. **automation-kit** passes the necessary data (contract addresses, ABI, method parameters) to **web3-adapter-core**, which then interacts with the blockchain to complete the transaction.
+
+
+#### Example `web3-adapter-core` API usage in `automation-kit`:
+
+```typescript
+import { sendTransaction, contractCall } from 'web3-adapter-core';
+import { VaultFactoryABI } from './abis/VaultFactoryABI.json';
+
+// Method to predict the vault address
+async function predictVaultAddress(creator: string, vaultId: number) {
+    const result = await contractCall({
+        address: '0xVaultFactoryAddress',
+        abi: VaultFactoryABI,
+        method: 'predictDeterministicVaultAddress',
+        args: [creator, vaultId],
+        type: 'view'
+    });
+    return result;
+}
+```
+
+
+### User Setup Example:
+
+1. **Initialize Web3Adapter**: Users will initialize **web3-adapter-core** using a configuration method like `init` or `use`. This setup allows users to choose which network library to use (e.g., ethers.js or web3.js).
+
+```typescript
+import { Web3Adapter } from 'web3-adapter-core';
+import { EthersAdapter } from 'web3-adapter-ethers';  // or web3.js
+
+// Initialize Web3Adapter with Ethers.js as the underlying adapter
+Web3Adapter.use(new EthersAdapter({
```

Comment:
https://docs.reown.com/appkit/react/core/installation?platform=ethers
better to use the same interface as walletconnect